docs/standard/datetime/system-text-json-support.md

```diff
@@ -186,6 +186,8 @@ The following levels of granularity are defined for parsing:
     3. "yyyy'-'MM'-'dd'T'HH':'mm':'ss('+'/'-')HH':'mm"
     4. "yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'FFFFFFF('+'/'-')HH':'mm"
 
+    This level of granularity is compliant with [RFC 3339](https://tools.ietf.org/html/rfc3339#section-5.6), a widely adopted profile of ISO 8601 used for interchanging date and time information.
```

Comment:
> For parsing, RFC 3339 seems to allow arbitrary many fractional seconds (at least 1).

Our parser supports up to 16, and only the first 7 are meaningful. Does it make sense to allow any arbitrary number of digits and not throw for those?

We allow up to 16 digits to be present to better interop with other languages. Java for instance supports up to 9 fractional digits in their DateTime formatting implementation: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html (search for "Fraction: Outputs the nano-of-second" on the page). I think our current behavior here makes sense. Any change here would be breaking.

> maybe it is worth highlighting that a bit more (since that seems to be the only difference). It's mentioned below, but adding a sentence here to connect it could be useful to some.

Thanks. I'll add a note about how there are some restrictions in the System.Text.Json implementation.

> how can I write a converter to allow that?

You can write a custom converter by copying the internal parsing logic in the reader and modifying the section that processes the fractional digits.

Reply:
Gotcha. And what about:

> will any other input string that satisfies RFC 3339 result in JsonSerializer.Deserialize throwing? Space delimiter, capital/lower case Z/T, etc.